Fix build regressions for `orjson`, `cramjam`, and PyXIRR due to Maturin 1.8

agriyakhetarpal opened the pull request:
<!-- Thank you for contributing to Pyodide! All improvements are welcome,
     so don't be afraid to make a PR. -->

### Description

<!-- Please explain what your PR is about:
     - reasoning for the change
     - some details of updated code
     - any noteworthy choices to be aware of
	Please refer to any related issues by #<issue_id> -->

Through https://github.com/PyO3/maturin/pull/2391 via the recently released [Maturin version 1.8.0](https://github.com/PyO3/maturin/releases/tag/v1.8.0), builds for PyXIRR (added in #4513) and `orjson` (added in #4036) have been failing.

Builds for other packages such as `cramjam` have been failing as well due to the same reason.

I've applied this PR for PyXIRR, which I submitted upstream: https://github.com/Anexen/pyxirr/pull/64, as a patch here, which can be dropped later. Tagging @Anexen as the listed recipe maintainer for visibility.

For `orjson`, bumping to the [newly released version 3.10.13](https://github.com/ijl/orjson/releases/tag/3.10.13) should work. Tagging @spicy-sauce as the listed recipe maintainer for visibility, no action needed.

### Checklists

<!-- Note:
     If you think some of these steps are not necessary for your PR,
     remove those checkboxes, or mark them as checked. If you keep unchecked checkboxes,
     we will assume that your PR is not ready to be merged  -->

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry

agriyakhetarpal committed:
Fix PyXiRR build regression

agriyakhetarpal commented:
Ah, `orjson` has the same problem, for which they've released a new version already: https://github.com/ijl/orjson/releases/tag/3.10.13

agriyakhetarpal committed:
Bump `orjson` to 3.10.13

agriyakhetarpal committed:
Add a CHANGELOG entry for #5283

spicy-sauce reviewed (changes requested):
check why the circleci checks fail...

hoodmane commented:
```
💥 maturin failed                                                               
  Caused by: `project.version` field is required in pyproject.toml unless it is 
present in the `project.dynamic` list                                           
Error: command ['maturin', 'pep517', 'build-wheel', '-i',                       
'/tmp/build-env-8rechu2o/bin/python', '--compatibility', 'off'] returned        
non-zero exit status 1
```

agriyakhetarpal commented:
Ah, now `cramjam` has the same error. I guess I will need to to do this for all Maturin-based packages.

agriyakhetarpal commented:
Submitted a PR to `cramjam` here: https://github.com/milesgranger/cramjam/pull/196

agriyakhetarpal committed:
Add patch for `cramjam`

agriyakhetarpal commented:
Argh, we have 19 packages that require Rust/maturin...

agriyakhetarpal committed:
Bump `pyxel` to version 2.2.10

agriyakhetarpal commented:
A short, internal tracker:

| Package | Affected? |
|--------|--------|
| bcrypt | No, it specifies a version and doesn't use `maturin` |
| cbor-diag | Yes |
| clarabel | Yes |
| cramjam | Will be fixed via https://github.com/milesgranger/cramjam/pull/196 and updated here |
| cryptography | No |
| lakers-python | Yes |
| libcst | No, uses `setuptools-rust` |
| nh3 | Yes |
| orjson | Yes, fixed here via https://github.com/ijl/orjson/releases/tag/3.10.13 |
| pyxel | Yes, fixed via https://github.com/kitao/pyxel/releases/tag/v2.2.10 and updated here |
| pyxirr | Yes, will be fixed via https://github.com/Anexen/pyxirr/pull/64 and updated here  |
| rateslib | No |
| river | No |
| rpds-py | Yes |
| rust-abi-test | No |
| rust-panic-test | No |
| sourmash | No |
| tiktoken | No |
| river | No |

hoodmane commented:
In the meantime can we cap maturin?

agriyakhetarpal commented:
Yes, makes sense to me!

ryanking13 commented:
Thanks for handling this issue!
